app/src/main/java/org/mozilla/fenix/search/toolbar/SearchSelectorToolbarAction.kt

```diff
@@ -71,19 +58,41 @@ class SearchSelectorToolbarAction(
         }
     }
 
-    override fun bind(view: View) = Unit
+    override fun bind(view: View) {
+        // It may happen that this View is binded multiple times.
+        // Prevent launching new coroutines for every time this is binded and only update the icon once.
+        if (updateIconJob?.isActive != true) {
+            updateIconJob = (view as? SearchSelector)?.toScope()?.launch {
```

Comment:
Used this form 
> CoroutineScope.launch { store.flow() .. }

instead of

> View.consumeFrom(Store, LifecycleOwner, (State))

more to just have a flow on which we can easily use the `ifChanged` operator and not having to write it ourselves.